src/libpsl-native/src/getppid.cpp

```diff
@@ -0,0 +1,39 @@
+#include "getppid.h"
+
+#include <sys/user.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+
+//! @brief GetPPid returns the parent process id for a process
+//!
+//! GetPPid
+//!
+//! @param[in] pid
+//! @parblock
+//! The process id to query for it's parent.
+//! @endparblock
+//!
+//! @retval the parent process id, or UINT_MAX if unsuccessful
+//!
+pid_t GetPPid(pid_t pid)
+{
+
+#if defined(__APPLE__) && defined(__MACH__)
+
+    const pid_t PIDUnknown = UINT_MAX;
+    struct kinfo_proc info;
+    size_t length = sizeof(struct kinfo_proc);
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, pid};
+    if (sysctl(mib, 4, &info, &length, NULL, 0) < 0)
+        return PIDUnknown;
+    if (length == 0)
+        return PIDUnknown;
+
+    return info.kp_eproc.e_ppid;
+
+#else
+
+    return UINT_MAX;
```

Comment:
Perhaps we should simply not compile this `GetPPid` function at all when not on Mac (i.e., move the `#if defined ... #endif` to surround the function definition and add one around the declaration). I'm honestly not sure which would be more maintainable: a "usable" (but not implemented) function on a platform always returning an invalid PID, or a runtime error from PowerShell not finding the symbol on attempted use because the function was never compiled; @lzybkr what do you think?